packages/exo/src/exo-tools.js

```diff
@@ -439,6 +427,21 @@ export const defendPrototype = (
     );
   }
 
+  if (interfaceGuard && !hasOwnPropertyOf(prototype, GET_INTERFACE_GUARD)) {
```

Comment:
I still don't like the first part of this conditional (`if (interfaceGuard &&`), since it means the return type of defendPrototype is not consistent with the static type of the return value (`/** @type {T & import('./get-interface.js').GetInterfaceGuard<T> */`).  This is no longer a sniffing problem, since you solved that with GET_METHOD_NAMES, but it is a Typescript problem (inconsistent return value that makes the static type incorrect).

Could you just do this condition based on `hasOwnPropertyOf`, but not based on the truthiness of `interfaceGuard`?

Reply:
Frankly, I'm not sure this is better. The runtime behavior has weaker invariants, since the method may exist but return `undefined`. Either way, we needed to weaken the static types. But had we weakened the static type of `defendPrototype` instead, we would not have needed to weaken the runtime invariants.

But in any case, done.